lib/gruf/client.rb

```diff
@@ -216,5 +216,25 @@ def error_deserializer_class
         Gruf::Serializers::Errors::Json
       end
     end
+
+    ##
+    # Handle various timeout values and prevent improper value setting
+    #
+    # @see GRPC::Core::TimeConsts#from_relative_time
+    # @param [mixed] timeout
+    # @return [Float]
+    # @return [GRPC::Core::TimeSpec]
+    #
+    def parse_timeout(timeout)
+      if timeout.nil?
+        GRPC::Core::TimeConsts::ZERO
+      elsif timeout.is_a?(GRPC::Core::TimeSpec)
+        timeout
+      elsif timeout.is_a?(Numeric)
+        timeout
+      else
+        timeout.to_f
+      end
```

Comment:
What if the timeout does not respond to `to_f`?

Reply:
Good call, probably need to throw an ArgumentError, rather than let the standard exception be raised